FESS-GUI/commoninterfaceselector.h

```diff
@@ -0,0 +1,39 @@
+#ifndef COMMONINTERFACESELECTOR_H
+#define COMMONINTERFACESELECTOR_H
+
+#include <QDialog>
+#include <QSerialPortInfo>
+
+#include "commoninterfacemanager.h"
+
+namespace Ui { class CommonInterfaceSelector;}
+
+class CommonInterfaceSelector : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit CommonInterfaceSelector(CommonInterfaceManager*, QWidget *parent = 0);
+    ~CommonInterfaceSelector();
+
+private slots:
+    void demoButtonSetClicked();
+    void demoButtonCancelClicked();
+
+    void serialButtonSetClicked();
+    void serialButtonCancelClicked();
+
+private:
+    Ui::CommonInterfaceSelector *ui;
+
+    bool serial_port_text_modified;
```

Comment:
I typically use lower case and underscores for build in data types and camelCase for objects but I can change it to camelCase to be consistent with the rest of the project.